external-crates/move/crates/move-package-alt/DESIGN.md

```diff
@@ -642,13 +633,68 @@ published-at = "TODO: replace this with the published address of your package"
 The user can still screw this up, but they at least have the chance of noticing the problem and
 fixing it when they review the commit. We can also output a loud warning from the command line!
 
+## Ephemeral publish
+
+A common operation that we would like to support is to make an ephemeral publication of a package
+and its dependencies for testing (for example on localnet). To support this, we will add an
+ephemeral publish command that works like a normal publish except that the publication addresses for
+packages are read from and written to a separate file.
+
+The format of an ephemeral publication file is as follows:
+
+```toml
+# generated by move
+# this file contains metadata from ephemeral publications
+# this file should not be committed to source control
+
+build-env = "mainnet"
+chain-id = "localnet chain ID"
+
+[published.example]
+published-at = "0x000000000000000000000000000000000000000000000000000000000000cccc"
+original-id = "0x000000000000000000000000000000000000000000000000000000000000cc00"
+upgrade-cap = "0x000000000000000000000000000000000000000000000000000000000011cc00"
+
+[published.Foo_0]
+published-at = "0x0000000000000000000000000000000000000000000000000000000000001234"
+original-id = "0x0000000000000000000000000000000000000000000000000000000000005678"
+upgrade-cap = "0x000000000000000000000000000000000000000000000000000000000022cc00"
+```
+
+We will add a new command `sui client test-publish <pubfile>` which will publish build the root
```

Comment:
I like the idea that these publishes are just considered "ephemeral" rather than necessarily for "test" purposes. It would be nice to indicate that in the command name as well. I don't have a good suggestion on the name, (`ephemeral-publish` seems like too annoying a thing to ask people to type out), but I'd like to table that bike-shedding topic.